packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m

```diff
@@ -74,108 +127,70 @@ - (void)testGetProductResponse {
   OCMStub([mockHandler addPayment:[OCMArg any]]).andReturn(NO);
   self.plugin.paymentQueueHandler = mockHandler;
 
-  [self.plugin handleMethodCall:call
-                         result:^(id _Nullable result) {
-                           FlutterError *error = result;
-                           XCTAssertEqualObjects(@"storekit_duplicate_product_object", error.code);
-                           XCTAssertEqualObjects(
-                               @"There is a pending transaction for the same product identifier. "
-                               @"Please either wait for it to be finished or finish it manually "
-                               @"using `completePurchase` to avoid edge cases.",
-                               error.message);
-                           XCTAssertEqualObjects(arguments, error.details);
-                           [expectation fulfill];
-                         }];
+  FlutterError *error;
+
+  [self.plugin addPayment:argument error:&error];
 
-  [self waitForExpectations:@[ expectation ] timeout:5];
   OCMVerify(times(1), [mockHandler addPayment:[OCMArg any]]);
+  XCTAssertEqualObjects(@"storekit_duplicate_product_object", error.code);
+  XCTAssertEqualObjects(@"There is a pending transaction for the same product identifier. "
+                        @"Please either wait for it to be finished or finish it manually "
+                        @"using `completePurchase` to avoid edge cases.",
+                        error.message);
+  XCTAssertEqualObjects(argument, error.details);
 }
 
 - (void)testAddPaymentSuccessWithoutPaymentDiscount {
-  XCTestExpectation *expectation =
-      [self expectationWithDescription:@"Result should return success state"];
-  FlutterMethodCall *call =
-      [FlutterMethodCall methodCallWithMethodName:@"-[InAppPurchasePlugin addPayment:result:]"
-                                        arguments:@{
-                                          @"productIdentifier" : @"123",
-                                          @"quantity" : @(1),
-                                          @"simulatesAskToBuyInSandbox" : @YES,
-                                        }];
+  NSDictionary *argument = @{
+    @"productIdentifier" : @"123",
+    @"quantity" : @(1),
+    @"simulatesAskToBuyInSandbox" : @YES,
+  };
+
   FIAPaymentQueueHandler *mockHandler = OCMClassMock(FIAPaymentQueueHandler.class);
   OCMStub([mockHandler addPayment:[OCMArg any]]).andReturn(YES);
   self.plugin.paymentQueueHandler = mockHandler;
-  [self.plugin handleMethodCall:call
-                         result:^(id _Nullable result) {
-                           XCTAssertNil(result);
-                           [expectation fulfill];
-                         }];
-  [self waitForExpectations:@[ expectation ] timeout:5];
+
+  FlutterError *error;
+
+  [self.plugin addPayment:argument error:&error];
+  XCTAssertNil(error);
+  OCMVerify(times(1), [mockHandler addPayment:[OCMArg any]]);
```

Comment:
Is it possible to add a matcher here for the argument to make sure it has the right product identifier and quantity?